openerp/addons/base/res/ir_property.py

```diff
@@ -91,68 +76,68 @@ def _update_values(self, cr, uid, ids, values):
+        elif self.type == 'integer':
+            return self.value_integer
+        elif self.type == 'binary':
+            return self.value_binary
+        elif self.type == 'many2one':
+            if not self.value_reference:
                 return False
-            model, resource_id = record.value_reference.split(',')
-            value = self.pool[model].browse(cr, uid, int(resource_id), context=context)
-            return value.exists()
-        elif record.type == 'datetime':
-            return record.value_datetime
-        elif record.type == 'date':
-            if not record.value_datetime:
+            model, resource_id = self.value_reference.split(',')
+            return self.env[model].browse(int(resource_id)).exists()
+        elif self.type == 'datetime':
+            return self.value_datetime
+        elif self.type == 'date':
+            if not self.value_datetime:
                 return False
-            return time.strftime('%Y-%m-%d', time.strptime(record.value_datetime, '%Y-%m-%d %H:%M:%S'))
+            return fields.Date.to_string(fields.Datetime.from_string(self.value_datetime))
         return False
 
-    def get(self, cr, uid, name, model, res_id=False, context=None):
-        domain = self._get_domain(cr, uid, name, model, context=context)
+    @api.model
+    def get(self, name, model, res_id=False):
+        domain = self._get_domain(name, model)
         if domain is not None:
-            domain = [('res_id', '=', res_id)] + domain
+            domain += [('res_id', '=', res_id)]
             #make the search with company_id asc to make sure that properties specific to a company are given first
-            nid = self.search(cr, uid, domain, limit=1, order='company_id asc', context=context)
-            if not nid: return False
-            record = self.browse(cr, uid, nid[0], context=context)
-            return self.get_by_record(cr, uid, record, context=context)
+            prop = self.search(domain, limit=1, order='company_id')
+            if prop:
+                return prop.get_by_record()
         return False
 
-    def _get_domain(self, cr, uid, prop_name, model, context=None):
-        context = context or {}
-        cr.execute('select id from ir_model_fields where name=%s and model=%s', (prop_name, model))
-        res = cr.fetchone()
-        if not res:
-            return None
-
-        cid = context.get('force_company')
-        if not cid:
-            company = self.pool.get('res.company')
-            cid = company._company_default_get(cr, uid, model, res[0], context=context)
+    def _get_domain(self, prop_name, model):
+        field = self.env['ir.model.fields'].search([('name', '=', prop_name), ('model', '=', model)], limit=1)
+        if not field: return None
 
-        return [('fields_id', '=', res[0]), ('company_id', 'in', [cid, False])]
+        company_id = self.env.context.get('force_company', self.env['res.company']._company_default_get().id)
```

Comment:
Use an `or` in case someone puts `force_company` to `False` in context:

    company_id = self._context.get('force_company') or self.env['res.company']._company_default_get(model).id